calculator.rb

```diff
@@ -0,0 +1,90 @@
+# prints a welcome message to the user
+print "Welcome to the calculating program! which operater you want to use \n \
+1. add(+)\n 2. substract(-)\n 3. multiply(*)\n 4. divide(/)\n"
+
+# prompt user to enter a math operater
+print "Please enter a math operater (name or a symbol): "
+math_operater = gets.chomp.downcase
+
+# creates an array with list of maths operator
+math_operators = ["+", "-", "/", "*", "add", "subtract", "multiply", "divide"]
+
+# ask user to input an maths operater until its valid
+until math_operators.include?(math_operater)
+    puts"Invalid math operator! Please try again."
+    print "Please enter a math operater: "
+    math_operater = gets.chomp.downcase
+end
+
+# rescue clause catches the argument error 
+begin
+    print "Please enter the first number: "
+    num_one = Integer(gets.chomp) 
+rescue  ArgumentError  => e
+    puts "You did not enter a a number. Try again!"
+end
```

Comment:
Hm, I'm looking at this above `begin ... rescue` section. You end up having this exact logic down below, inside of an `until` loop... What happens if we replace this above logic with this?

```ruby
print "Please enter the first number: "
num_one = Integer(gets.chomp) 
```

```ruby